pkg/reconciler/bitbucketsource/webhook_client.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bitbucketsource
+
+import (
+	"context"
+	"fmt"
+	bbclient "github.com/knative/eventing-sources/pkg/reconciler/bitbucketsource/resources"
+	"github.com/knative/pkg/logging"
+	"golang.org/x/oauth2/bitbucket"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+type webhookOptions struct {
+	uuid           string
+	consumerKey    string
+	consumerSecret string
+	domain         string
+	owner          string
+	repo           string
+	events         []string
+}
+
+type webhookClient interface {
+	Create(ctx context.Context, options *webhookOptions) (string, error)
+	Delete(ctx context.Context, options *webhookOptions) error
+}
+
+type bitBucketWebhookClient struct{}
+
+func (client bitBucketWebhookClient) Create(ctx context.Context, options *webhookOptions) (string, error) {
+	logger := logging.FromContext(ctx)
+
+	logger.Info("Creating BitBucket WebHook")
+
+	bbClient, err := client.createBitBucketClient(ctx, options)
+
+	if err != nil {
+		return "", err
+	}
+
+	hook := client.hookConfig(options)
+
+	var h *bbclient.Hook
+	h, err = bbClient.CreateHook(options.owner, options.repo, &hook)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to Create the BitBucket Webhook: %v", err)
+	}
+	logger.Infof("Created BitBucket WebHook: %+v", h)
+	return h.UUID, nil
+}
+
+func (client bitBucketWebhookClient) Delete(ctx context.Context, options *webhookOptions) error {
+	logger := logging.FromContext(ctx)
+
+	logger.Info("Deleting BitBucket WebHook: %q", options.uuid)
+
+	bbClient, err := client.createBitBucketClient(ctx, options)
+
+	if err != nil {
+		return err
+	}
+
+	err = bbClient.DeleteHook(options.owner, options.repo, options.uuid)
+
+	if err != nil {
+		return fmt.Errorf("failed to Delete the BitBucket Webhook: %v", err)
+	}
+
+	logger.Infof("Deleted BitBucket Webhook: %s", options.uuid)
+	return nil
+}
+
+func (client bitBucketWebhookClient) createBitBucketClient(ctx context.Context, options *webhookOptions) (*bbclient.Client, error) {
```

Comment:
`client` is not used in this method, which makes me think it should just be a package method.

Reply:
Done